.travis.yml

```diff
@@ -30,10 +31,21 @@ perl:
 #     on_failure: always
 env:
   global:
-    - AUTHOR_TESTING=1
     - AUTOMATED_TESTING=1
-    - RELEASE_TESTING=0
+matrix:
+  include:
+    - perl: stable
+      env: AUTHOR_TESTING=1 RELEASE_TESTING=1
+  allow_failures:
+    - perl: blead
+    - perl: dev
+    - perl: stable
+      env: AUTHOR_TESTING=1 RELEASE_TESTING=1
 before_install:
+  # * global defaults (allows project network-side overloads [for non-AUTHOR build flexibility])
+  - export AUTHOR_TESTING=${AUTHOR_TESTING:=0}
```

Comment:
This change doesn't make any sense to me. AUTHOR_TESTING is set in order to run tests in xt/author. It doesn't have anything to do with who is running the test.

Reply:
With the unchanged Travis CI, *all* testing with AUTHOR_TESTING=1 *fails* during the xt\... tests (see <https://travis-ci.org/rivy/perl.File-Temp/builds/370646998>). As you say, AUTHOR_TESTING enables those tests. I can't modify the xt tests because they are not even in the repository; they are pulled in with some magic by the travis helper scripts. So, as I said in the initial post, I modified the Travis CI config to be useful for testing for contributors while still allowing individual users to override the default null value in their own cloud settings.

And, generally, AUTHOR_TESTING and RELEASE_TESTING as reserved for local author use, especially prior to release. So, I'm not sure why you'd want that tested for each commit anyway.

I did leave in one "failable" job which tests under "stable" with both "AUTHOR_TESTING=1" and "RELEASE_TESTING=1" (see <https://travis-ci.org/rivy/perl.File-Temp/jobs/494642590>) in case that it was interesting to some author(s).

Reply:
> With the unchanged Travis CI, _all_ testing with AUTHOR_TESTING=1 _fails_ during the xt... tests (see https://travis-ci.org/rivy/perl.File-Temp/builds/370646998).

Then the answer is to fix the failing tests, not to remove those tests from being run. If you look, you'll see that it would be quite simple to make that test pass -- as it's pointing out a real problem with your PR.

Reply:
> So, I'm not sure why you'd want that tested for each commit anyway.

They are author tests because they do not need to run for the end-user installation -- because they are testing things that do not differ per-installation and should be corrected by the author before release.

In the case of this PR, *you are the author*.

Reply:
We have basic disagreement on terms and context.

It's your module, you being the or one of the authors as I see it ... so I'll try to change my context to yours.

But it's not simple and obvious. Again, the test is not in the repository, so I couldn't review the code to determine the failing issue. The error notation "#   Failed test 'Changes has content for 0.2306'" implied that there needed to be some entry in Changes for 0.2306, but no matter what I put into that file, it still failed. Here is a reproduction of an attempt (which fails), see <https://travis-ci.org/rivy/perl.File-Temp/builds/495178912>.

So, I came to the conclusion that, as a contributor, not a publishing author, I should not be testing those things and concocted the Travis in the manner and for the reasons I explained.

But, you want me to make the AUTHOR tests pass ... fine, tell me what the "simple" thing is that makes the test pass. And it might be nice to somehow communicate that to those who might want to contribute.

Reply:
> tell me what the "simple" thing is that makes the test pass

You just need to add an entry to the Changes file, below the placeholder `{{$NEXT}}` that is turned into the version+date at release time, just as in commit e2ad5b4d7.